metricbeat/module/elasticsearch/ml_job/ml_job.go

```diff
@@ -53,12 +55,19 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right format
 func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 	shouldSkip, err := m.ShouldSkipFetch()
+	mLAvailability, mlErr := m.checkMLAvailability()
```

Comment:
we can move this after line 64 so we don't make the request if not necessary

Reply:
Moved, thanks 👍